### PR TITLE
144/date input/uip 824 close on select

### PR DESCRIPTION
**Pull request by akdetrick**
## Description

Updates `DateInput` to use a controlled popover. We open the popover on input focus. The popover closes `onUserDismiss` _and_ on date selection.

I also took this opportunity to separate out helper components now that we have this nice v9 file organization

## Checklist
- [x] Docs have been rebuilt (`yarn build:full`)
- [x] All unit tests pass
- [x] Snapshots have been updated
- [x] No lint errors or warnings have been introduced
- [x] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: src/components/DateInput/dateInput.test.js [hunk 1]
```diff
@@ -3,155 +3,106 @@ import { render, screen, fireEvent } from '@testing-library/react'
...
+    fireEvent.click(screen.getByText('7'));
+    expect(dateChangeFn).toHaveBeenCalled();
+  });
 
-    it('does not call onDateChange when user types an INVALID freeform date', async () => {
-      expect(dateChangeFn).not.toHaveBeenCalled();
-      await userEvent.type(screen.getByRole('textbox'), '2/30/2020')
-      expect(dateChangeFn).not.toHaveBeenCalled();
-    });
+  it('updates inupt value when user selects a day in the picker', () => {
```
Comment (rey-wright) [inline src/components/DateInput/dateInput.test.js:54]:
```suggestion
  it('updates input value when user selects a day in the picker', () => {
```

File: src/components/DateInput/dateInput.test.js [hunk 2]
```diff
@@ -3,155 +3,106 @@ import { render, screen, fireEvent } from '@testing-library/react'
...
+  });
 
-    it('does not call onDateChange when user types an INCOMPLETE freeform date', async () => {
-      expect(dateChangeFn).not.toHaveBeenCalled();
-      await userEvent.type(screen.getByRole('textbox'), '2/3')
-      expect(dateChangeFn).not.toHaveBeenCalled();
-    });
+  it('calls onDateChange when user selects a day in the picker', () => {
+    expect(dateChangeFn).not.toHaveBeenCalled();
+    fireEvent.click(screen.getByText('7'));
```
Comment (rey-wright) [inline src/components/DateInput/dateInput.test.js:50]:
```suggestion
     fireEvent.click(screen.getByRole('gridcell', { name: 'Tue Jul 07 2020' }));
```

Comment (rey-wright) [reply src/components/DateInput/dateInput.test.js:50]: I should have updated these in my tests on master. You may have to rebase / merge but you shouldn't need to use `getByText` in any of the tests in this file.

Comment (akdetrick) [reply src/components/DateInput/dateInput.test.js:50]: did a fresh merge from master; should be resolved now

File: src/components/DateInput/dateInput.test.js [hunk 3]
```diff
@@ -3,155 +3,107 @@ import { render, screen, fireEvent } from '@testing-library/react'
...
```
Comment (rey-wright) [inline src/components/DateInput/dateInput.test.js:50]:
```suggestion
    fireEvent.click(screen.getByRole('gridcell', { name: 'Tue Jul 07 2020' }));
```

File: src/components/DateInput/dateInput.test.js [hunk 4]
```diff
@@ -3,155 +3,107 @@ import { render, screen, fireEvent } from '@testing-library/react'
...
+    expect(dateChangeFn).toHaveBeenCalled();
+  });
 
-    it('does not call onDateChange when user types an INVALID freeform date', async () => {
-      expect(dateChangeFn).not.toHaveBeenCalled();
-      await userEvent.type(screen.getByRole('textbox'), '2/30/2020')
-      expect(dateChangeFn).not.toHaveBeenCalled();
-    });
+  it('updates input value when user selects a day in the picker', () => {
+    fireEvent.click(screen.getByText('7'));
```
Comment (rey-wright) [inline src/components/DateInput/dateInput.test.js:55]:
```suggestion
    fireEvent.click(screen.getByRole('gridcell', { name: 'Tue Jul 07 2020' }));
```

**Pull request merged**